homeassistant/components/airzone_cloud/sensor.py

```diff
@@ -0,0 +1,97 @@
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+from .entity import AirzoneEntity, AirzoneZoneEntity
+
+ZONE_SENSOR_TYPES: Final[tuple[SensorEntityDescription, ...]] = (
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.TEMPERATURE,
+        key=AZD_TEMP,
+        name="Temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.HUMIDITY,
+        key=AZD_HUMIDITY,
+        name="Humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone Cloud sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = []
+    for zone_id, zone_data in coordinator.data[AZD_ZONES].items():
+        for description in ZONE_SENSOR_TYPES:
+            if description.key in zone_data:
+                sensors.append(
+                    AirzoneZoneSensor(
+                        coordinator,
+                        description,
+                        entry,
+                        zone_id,
+                        zone_data,
+                    )
+                )
+
+    async_add_entities(sensors)
+
+
+class AirzoneSensor(AirzoneEntity, SensorEntity):
+    """Define an Airzone Cloud sensor."""
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Update attributes when the coordinator updates."""
+        self._async_update_attrs()
+        super()._handle_coordinator_update()
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update sensor attributes."""
+        self._attr_native_value = self.get_airzone_value(self.entity_description.key)
+
+
+class AirzoneZoneSensor(AirzoneZoneEntity, AirzoneSensor):
+    """Define an Airzone Cloud Zone sensor."""
+
+    def __init__(
+        self,
+        coordinator: AirzoneUpdateCoordinator,
+        description: SensorEntityDescription,
+        entry: ConfigEntry,
+        zone_id: str,
+        zone_data: dict[str, Any],
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator, entry, zone_id, zone_data)
+
+        self._attr_name = f"{zone_data[AZD_NAME]} {description.name}"
+        self._attr_unique_id = f"{entry.entry_id}_{zone_id}_{description.key}"
```

Comment:
Here the entry.entry_id is used to derive the unique id, but in the base class the config entry's unique id is preferred.

Reply:
Just to be sure, you mean that I should do `self._attr_unique_id = f"{entry.unique_id}_{zone_id}_{description.key}"` here, right?

Reply:
Changed in c779c6473cbdc94e7d411f143675dca6c0c0c053